src/main/java/com/demonwav/mcdev/buildsystem/gradle/GradleBuildSystem.java

```diff
@@ -305,15 +307,18 @@ public void finishSetup(@NotNull Module module, @NotNull ProjectConfiguration co
                 // Set the Forge client and server run configs
                 if (configuration.type == PlatformType.FORGE) {
                     // Client run config
-                    ExternalSystemRunConfiguration runClientConfiguration = new ExternalSystemRunConfiguration(
-                            GradleConstants.SYSTEM_ID,
+                    ApplicationConfiguration runClientConfiguration = new ApplicationConfiguration(
+                            module.getName() + " run client",
                             project,
-                            gradleType.getConfigurationFactories()[0],
-                            module.getName() + " run client"
+                            ApplicationConfigurationType.getInstance()
                     );
-                    runClientConfiguration.getSettings().setExternalProjectPath(rootDirectory.getPath());
-                    runClientConfiguration.getSettings().setExecutionName(module.getName() + " run client");
-                    runClientConfiguration.getSettings().setTaskNames(Collections.singletonList("runClient"));
+                    File runningDir = new File(project.getBasePath(), "run");
+                    if (!runningDir.exists()) {
+                        runningDir.mkdir();
+                    }
+                    runClientConfiguration.setWorkingDirectory(project.getBasePath() + File.separator + "run");
+                    runClientConfiguration.setMainClassName("GradleStart");
+                    runClientConfiguration.setModule(ModuleManager.getInstance(project).findModuleByName(module.getName() + "_main") != null ? ModuleManager.getInstance(project).findModuleByName(module.getName() + "_main") : module);
```

Comment:
Don't do the module lookup twice. Also, only go past the 140 column line limit if breaking would hurt readability.